sampleconfig/orderer.yaml

```diff
@@ -361,6 +361,9 @@ ChannelParticipation:
     # The maximum size of the request body when joining a channel.
     MaxRequestBodySize: 1 MB
 
+    # Defines the file repo directory where channel partipation resources are stored.
+    FileRepoDir: /var/hyperledger/production/orderer/filerepo
```

Comment:
Do we really need to make it configurable? What is the advantage of doing this? Too many configurations affect the usability IMO. 

In the peer, we have a single file path configuration `fileSystemPath` and create all necessary internal folders there. For the ledger snapshot, a new filePath configuration would be introduced so that the user can decide to keep the snapshot on different storage especially when they create a snapshot to perform archival as the  `fileSystemPath` has no free capacity. 

If there is no such reasoning for making this configurable, I would prefer to remove this. Moreover, one needs to understand the internal of orderer before making a configuration decision on `FileRepoDir`, etc... IMO, this is internal to the orderer logic.

Reply:
I think I agree with you here. The initial decision on configuring the file repo dir was from before when there was consideration for multiple file repo dirs (ie joinblock and remove). But after some refactoring in the file repo PR we've switched to subfolders of a root file repo parent dir (also reduces the number of configurations for needing to have one for each file repo dir). I can go ahead and remove this as it also simplifies some of the logic around requiring the parent repo dir to exist if we just create it internally as part of the orderer startup and default to an internal orderer subfolder.

Reply:
@cendhu do you have any objections with reusing the file ledger's location so we don't have to create additional paths? Ie based off the default for the file ledger's location we could internally create a folder at `/var/hyperledger/production/orderer/filerepo` which is prefixed by the file ledger location.

Reply:
The consensus `WALDir` and `SnapDir` seem to be configurable but the defaults are based off of the `FileLedger.Location` so that's probably the closest to a `fileSystemPath` for the orderer

Reply:
Ah nevermind it seems the `FileLedger.Location` isn't the best choice here as it creates a temporary folder on orderer restart if the location is not configured. @cendhu what are your thoughts on the orderer's nonexistent `fileSystemPath` for internal folders. There doesn't seem to be a concept for one in the existing configuration.

Reply:
I think mirroring the peer here and adding a configurable `fileSystemPath` probably makes the most sense.

Reply:
As discussed offline, @manish-sethi, @caod123 and I are in favor of removing the temp file logic related to 
`FileLedger.Location`

> NOTE: If this is unset, a new temporary location will be chosen every time
> the orderer is restarted, using the prefix specified by Prefix.

Reply:
https://github.com/hyperledger/fabric/pull/1598 was recently merged to address this discussion. I'll update this PR soon to incorporate that and remove this FileRepoDir configuration option.